### PR TITLE
Change the Implimentation of CSRMatrix to us SparseMatrixCSC

### PR DESCRIPTION
**Pull request by ndinsmore**
This is a continuation of #96.    Which got accidentally closed.
The gist of this is that it changes CSRMatrix to be an alias for Transpose{SparseMatrixCSC}

### CODE REVIEW

**Comment by joaquimg** [comment]
So there is a great improvement on Clp, right? Have you benchmarked any other solvers?

**Comment by ndinsmore** [comment]
This change it's self isn't really responsible for much performance improvement, it is more of a move towards base julia functionality.   But it does result in on due to SparseMatrixCSC haveing many performance optimizations.   

The benchmarks for this change were neutral because it effectively is just code simplification.  

I do have the changes ready for Gurobi.jl, and GLPK.jl.   Both of which are literally a few characters.   I was just waiting for this to get merged before submitting them.

File: src/constraints.jl [hunk 1]
```diff
@@ -160,31 +160,29 @@ end
...
-`CSRMatrix` is analgous to the structure in Julia's `SparseMatrixCSC` but with
-the rows and columns flipped. It contains three vectors:
- - `row_pointers` is a vector pointing to the start of each row in
-    `columns` and `coefficients`;
- - `columns` is a vector of column numbers; and
- - `coefficients` is a vector of corresponding nonzero values.
+Type alias for `Transpose{T,SparseMatrixCSC{T,Int}} where T`, which allows for the
+    efficient storage of compressed sparse row (CSR) sparse matrix
+
+Please use the following interface to access this data incase model is changed in future
```
Comment (blegat) [inline src/constraints.jl]: incase -> in case

**Comment by ndinsmore** [comment]
Are there any blocking issues or concerns from the reviewers?

**Comment by joaquimg** [comment]
If there are no regressions on gurobi I am ok with merging.
It would be great to see some benchmark though.

**Comment by ndinsmore** [comment]
@joaquimg I don't have access to gurobi, but GLPK is similar in use and the benchmarks in https://github.com/JuliaOpt/GLPK.jl/pull/95 show no change in performance.

**Comment by joaquimg** [comment]
Then its good to go IMO.

**Comment by odow** [comment]
Can we rebase this on the latest master please? Then it's good to merge.

**Comment by ndinsmore** [comment]
I think that we are good to go.

**Pull request merged**